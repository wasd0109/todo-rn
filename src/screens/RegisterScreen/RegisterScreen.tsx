import React, { useState } from "react";
import { StyleSheet, Text, View } from "react-native";
import StyledTextInput from "../../components/StyledTextInput";
import StyledSecureInput from "../../components/StyledSecureInput";
import { Button } from "react-native-paper";
import Spacer from "../../components/Spacer";

const NAME_PLACEHOLDER = "John Smith";
const EMAIL_PLACEHOLDER = "abc@gmail.com";
const PASSWORD_PLACEHOLDER = "******";

const RegisterScreen = () => {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  return (
    <View style={styles.containerStyle}>
      <StyledTextInput
        label="Name"
        placeholder={NAME_PLACEHOLDER}
        value={name}
        onChangeText={setName}
        testID="name-input"
      />
      <StyledTextInput
        label="Email"
        placeholder={EMAIL_PLACEHOLDER}
        value={email}
        onChangeText={setEmail}
        testID="email-input"
      />
      <StyledSecureInput
        label="Password"
        placeholder={PASSWORD_PLACEHOLDER}
        value={password}
        onChangeText={setPassword}
        testID="password-input"
      />
      <Spacer>
        <View style={styles.btnContainerStyle}>
          <Button style={styles.btnStyle} mode="contained">
            Register
          </Button>
          <Button style={styles.btnStyle}>Login</Button>
        </View>
      </Spacer>
    </View>
  );
};

export default RegisterScreen;

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    marginTop: 200,
  },
  btnStyle: {
    width: 150,
  },
  btnContainerStyle: {
    flexDirection: "row",
    flex: 1,
    justifyContent: "space-evenly",
  },
});
