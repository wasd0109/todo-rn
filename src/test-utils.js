import React from "react";
import "react-native-paper";
import { render } from "@testing-library/react-native";
import { Provider as PaperProvider } from "react-native-paper";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";

const AllTheProviders = ({ children }) => {
  return <PaperProvider>{children}</PaperProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react-native";

// override render method
export { customRender as render };

const Stack = createStackNavigator();
const MockedNavigator = ({ screenName, component, params = {} }) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name={screenName}
          component={component}
          initialParams={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockedNavigator;
