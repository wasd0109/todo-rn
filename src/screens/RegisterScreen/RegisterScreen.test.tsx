import { RenderAPI } from "@testing-library/react-native";
import React from "react";
import { render, fireEvent } from "../../test-utils";
import RegisterScreen from "./RegisterScreen";

const TEST_NAME = "John Ele";
const TEST_EMAIL = "johnEle@gmail.com";
const TEST_PASSWORD = "testpassword";

let component: RenderAPI;

beforeEach(() => {
  component = render(<RegisterScreen />);
});

describe("RegisterScreen render correctly", () => {
  test("RegisterScreen render", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("RegisterScreen show Name Input", () => {
    const { getAllByTestId } = component;
    expect(getAllByTestId("name-input").length).toEqual(1);
  });

  test("RegisterScreen show Email Input", () => {
    const { getAllByTestId } = component;
    expect(getAllByTestId("email-input").length).toEqual(1);
  });

  test("RegisterScreen show Password Input", () => {
    const { getAllByTestId, getByTestId } = component;
    const passwordInputRef = getByTestId("password-input");
    console.log(passwordInputRef);
    expect(getAllByTestId("password-input").length).toEqual(1);
  });

  test("RegisterScreen show Register Button", () => {
    const { getAllByText } = component;
    expect(getAllByText("Register").length).toEqual(1);
  });

  test("RegisterScreen show Link to Login", () => {
    const { getAllByText } = component;
    expect(getAllByText("Login").length).toEqual(1);
  });
});

describe("Register Screen functions properly", () => {
  test("Name input allow input and changes value", () => {
    const { getByTestId, getAllByDisplayValue } = component;
    const nameInputRef = getByTestId("name-input");
    fireEvent.changeText(nameInputRef, TEST_NAME);
    expect(getAllByDisplayValue(TEST_NAME).length).toEqual(1);
  });

  test("Email input allow input and changes value", () => {
    const { getByTestId, getAllByDisplayValue } = component;
    const emailInputRef = getByTestId("email-input");
    fireEvent.changeText(emailInputRef, TEST_EMAIL);
    expect(getAllByDisplayValue(TEST_EMAIL).length).toEqual(1);
  });

  test("Password input allow input and changes value", () => {
    const { getByTestId, getAllByDisplayValue } = component;
    const passwordInputRef = getByTestId("password-input");
    fireEvent.changeText(passwordInputRef, TEST_PASSWORD);
    expect(getAllByDisplayValue(TEST_PASSWORD).length).toEqual(1);
  });
});
