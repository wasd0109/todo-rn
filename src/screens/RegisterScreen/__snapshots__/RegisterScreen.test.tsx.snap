// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterScreen render correctly RegisterScreen render 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <Text />
  </View>
</View>
`;
